libdnf/dnf-repo.c

```diff
@@ -1215,7 +1339,13 @@ dnf_repo_check_internal(DnfRepo *repo,
         if (!dnf_repo_set_timestamp_modified(repo, error))
             return FALSE;
         age_of_data =(g_get_real_time() - priv->timestamp_modified) / G_USEC_PER_SEC;
-        if (age_of_data > permissible_cache_age) {
+        metadata_expire = dnf_repo_get_metadata_expire(repo);
+        if(metadata_expire != 0)
+            valid_time_allowed = metadata_expire <= permissible_cache_age ? metadata_expire  : permissible_cache_age;
```

Comment:
hmm, I agree it is a bit redundant to have both attribute at the same time, but removing the permissible_cache_age( I feel) in some ways will eliminate the usage of the attribute cache_age for dnf_context. 

Before what we do(from what I observe) to set the cache age is to directly get a context object and set it through dnf_context related function calls.  Then we use that cache attribute as permissible cache age  to tell whether the metadata is expired  

If we were to adopt the metadata_expire and remove the permissible_cache_age,  it seems like the usage for cache age from dnf_context is no longer meaningful. 

As a result, even though I agree it makes sense to have only one, but the process of refactoring may be a bit more painful than the benefit we gain.

Again, all of it is just my personal thoughts, some of it could be completely wrong :p, if they were, please let me know :D. 

P.S: I am fine with doing the refactoring too if it is needed :D.